### DISCUSSION TITLE
Clarifying details of event/recording start and stop

### DISCUSSION THREAD

**Discussion by tgiffney** [blakeblackshear/frigate#4445] [Ask A Question] [answered]
Hi, thanks for the great software! I have some questions about the details of recording only during events.

1. How does the config option `pre_capture` interact with saving recording segments from the cache, when recording is only enabled for events (no `all` days of recording retained)? (will Frigate know to save the extra footage needed for however many seconds of `pre_capture` in the cache, or will the relevant segments unfortunately be discarded before pre_capture can have an effect)
2. Events begin if the bottom center of the bounding box enters the `required_zone`. Is the event supposed to end if the bottom center of the bounding box exits the `required_zone`?
3. The documentation section "Reducing false positives" talks about computing a score as the median of a score history, at least 3 values. Is there a maximum number of values considered in this median?

Thanks for your help!

**Comment by NickM-27** [comment, marked as answer]
1. There's multiple factors here. For one it depends on your events retention mode, pre_capture will not override that. And I believe multiple segments are kept in cache so pre_capture should still apply for events only. See https://github.com/blakeblackshear/frigate/issues/2270#issuecomment-1065609809

2. Events begin when the object is detected. If `required_zones` is set then that object must enter a zone for the event to be saved. But once the object enters a zone, the entire time the object was visible is a part of that event. Similarly, the event continues until that object is no longer detected. 

3. ~~It is the last 3 values, not 3+ values~~it is the last 10 values, it is a rolling median.

**Reply by blakeblackshear** [reply]
> It is the last 3 values, not 3+ values, it is a rolling median

It's actually a rolling 10 values. It just starts calculating once there are 3 values.

**Reply by NickM-27** [reply]
oh okay good to know, thanks!
